source/features/preserve-file-finder-term.tsx

```diff
@@ -0,0 +1,43 @@
+import features from '../libs/features';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+
+const FILE_FINDER_INPUT_SELECTOR = '.js-tree-finder > .breadcrumb > #tree-finder-field';
+
+const unloadHandler = (): void => {
+	sessionStorage.setItem('rgh-file-finder-term', select<HTMLInputElement>(FILE_FINDER_INPUT_SELECTOR)!.value);
+};
+
+// Set the input field value & trigger event
+const setValueInField = async (): Promise<void> => {
+	const value: string = sessionStorage.getItem('rgh-file-finder-term') ?? '';
+	const inputElement = select<HTMLInputElement>(FILE_FINDER_INPUT_SELECTOR);
+	if (inputElement && !inputElement.value && value) {
+		await elementReady('.js-tree-browser-results > li', {stopOnDomReady: false});	// For search to work
+		inputElement.value = value;
+		inputElement.dispatchEvent(new Event('input')); // Manually trigger event to trigger search
+	}
+};
+
+async function init(): Promise<void> {
+	await setValueInField();
```

Comment:
It's needed for the `elementReady` function inside `setValueinField`. I'm waiting for the element tree to be ready because we need a list of files when the input is populated with the value, for the filtering to take place. Otherwise the value is filled in, but files not filtered.

Reply:
That’s irrelevant; all that this await is doing is delaying the listeners that follow. That happens inside `setValueInField` isn’t touched by this await